src/sentry/options/defaults.py

```diff
@@ -2013,6 +2013,27 @@
     flags=FLAG_PRIORITIZE_DISK | FLAG_AUTOMATOR_MODIFIABLE,
 )
 
+# Enables split discover dataset task for dashboard widgets.
+register(
+    "split_discover_dataset.enable",
+    default=True,
+    flags=FLAG_PRIORITIZE_DISK | FLAG_AUTOMATOR_MODIFIABLE,
+)
+# Number of DashboardWidgets to be checked at once.
+register(
+    "split_discover_dataset.query.batch_size",
+    type=Int,
+    default=10,
```

Comment:
We should lower the batch number to keep the child task fast to not take up all the celery. I wrote [down some options here](https://www.notion.so/sentry/Splitting-Discover-Dataset-Task-81d7ce40958643cdb870c9e9a1307aff) as well as the redash query for entity count for estimates.